apps/deploy/upload-dapp-chain/src/dapp-data/dapp-data.service.ts

```diff
@@ -0,0 +1,217 @@
+
+    const gas = await gearApi.program.calculateGas.initUpload(
+      sourceId,
+      optWasmBuff,
+      meta.types,
+      0,
+      true,
+    );
+    const program = {
+      code: optWasmBuff,
+      gasLimit: gas.min_limit,
+      value: 0,
+    };
+    const { programId } = gearApi.program.upload(program, meta);
+
+    dappData.id = programId;
+    dappData.release = release;
+    dappData.metaWasmBase64 = metaWasmBuff.toString("base64");
+
+    const updateDapp = await this.dappDataRepository.save(dappData);
+
+    return { ...updateDapp, metaWasmBase64: "long" };
+  }
+
+  private async uploadNewDappInChain(uploadNewDappInChainInput: UploadNewDappInChainInput): Promise<DappData> {
+    const { sourceId, metaWasmBuff, optWasmBuff, gearApi, dappName, release, repo } = uploadNewDappInChainInput;
+    const paylaod = this.getPayloadByDappName(dappName);
+
+    if (!paylaod) {
+      throw new BadRequestExc("Invalid dapp name");
+    }
+
+    const meta = await getWasmMetadata(metaWasmBuff);
+
+    const gas = await gearApi.program.calculateGas.initUpload(
+      sourceId,
+      optWasmBuff,
+      meta.types,
+      0,
+      true,
+    );
+
+    const { programId } = gearApi.program.upload({
+      code: optWasmBuff,
+      gasLimit: gas.min_limit,
+      value: 0,
+      initPayload: this.getPayloadByDappName(dappName),
+    }, meta);
+
+    const createDappInput: CreateDappInput = {
+      id: programId,
+      name: dappName,
+      release,
+      repo,
+      metaWasmBase64: metaWasmBuff.toString("base64"),
+    };
+
+    const dapp = await this.createDapp(createDappInput);
+
+    return { ...dapp, metaWasmBase64: "long" };
+  }
+
+  private createDapp(createDappInput: CreateDappInput): Promise<DappData> {
+    const dappDbType = plainToClass(DappData, {
+      ...createDappInput,
+    });
+    return this.dappDataRepository.save(dappDbType);
+  }
+
+  private getDownloadUrlOptAndMetaWasmByNameDapp(
+    name: string,
+    assets: Asset[],
+  ): string[] {
+    const result = [];
+
+    for (const asset of assets) {
+      if (name === getNameDapp(asset.name) && asset.name.split(".").includes("opt")) {
+        result[0] = asset.browser_download_url;
+      }
+      if (name === getNameDapp(asset.name) && asset.name.split(".").includes("meta")) {
+        result[1] = asset.browser_download_url;
+      }
+    }
+    return result;
+  }
+
+  private getPayloadByDappName(name: string) {
+    const payloads = {
+      [DAPP.NFT]: { name: "test", symbol: "test", baseUri: "test", royalties: null },
+      [DAPP.NFT_MARKETPLACE]: { adminId: this.configService.get<string>("nftMarketplace.NFT_MARKETPLACE_ADMIN_ID"),
+        treasuryId: this.configService.get<string>("nftMarketplace.NFT_MARKETPLACE_TREASURY_ID"),
+        treasuryFee: "1" },
+      [DAPP.LOTTERY]: "0x00",
+      [DAPP.DUTCH_AUCTION]: "0x00",
+      [DAPP.ESCROW]: "--",
+      [DAPP.SUPPLY_CHAIN]: "--",
+      default: false,
```

Comment:
Let's take this out of the source code. I see a few ways how to do it.
1. Store payloads in database, and add new payloads for new apps manually (through tgbot).
2. Create `.yaml` files for each of the apps and read store it on the server.

Reply:
moved to `yaml` file